src/main/java/com/cloudogu/smeagol/ScmHttpClient.java

```diff
@@ -0,0 +1,108 @@
+import java.util.Optional;
+
+@Service
+public class ScmHttpClient {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ScmHttpClient.class);
+
+    private RestTemplate restTemplate;
+    private AccountService accountService;
+
+    @Autowired
+    public ScmHttpClient(AccountService accountService, RestTemplateBuilder restTemplateBuilder, @Value("${scm.url}") String scmUrl, Stage stage) {
+        this.accountService = accountService;
+
+        RestTemplateBuilder builder = restTemplateBuilder;
+        if (stage == Stage.DEVELOPMENT) {
+            builder = disableSSLVerification(restTemplateBuilder);
+        }
+        this.restTemplate = builder.rootUri(scmUrl).build();
+    }
+
+    private RestTemplateBuilder disableSSLVerification(RestTemplateBuilder restTemplateBuilder) {
+        LOG.warn("disable ssl verification for scm-manager communication, because we are in development stage");
+        TrustStrategy acceptingTrustStrategy = (X509Certificate[] chain, String authType) -> true;
+        SSLContext sslContext = createSSLContext(acceptingTrustStrategy);
+
+        SSLConnectionSocketFactory csf = new SSLConnectionSocketFactory(sslContext);
+        CloseableHttpClient httpClient = HttpClients.custom()
+                .setSSLSocketFactory(csf)
+                .build();
+
+        HttpComponentsClientHttpRequestFactory requestFactory =
+                new HttpComponentsClientHttpRequestFactory();
+
+        requestFactory.setHttpClient(httpClient);
+        return restTemplateBuilder.requestFactory(requestFactory);
+    }
+
+    private SSLContext createSSLContext(TrustStrategy acceptingTrustStrategy) {
+        try {
+            return SSLContexts.custom()
+                    .loadTrustMaterial(null, acceptingTrustStrategy)
+                    .build();
+        } catch (NoSuchAlgorithmException | KeyManagementException | KeyStoreException ex) {
+            throw Throwables.propagate(ex);
+        }
+    }
+
+    public <T> Optional<T> get(String url, Class<T> type, Object... urlVariables) {
+        return getEntity(url, type, urlVariables).map(e -> e.getBody());
+    }
+
+    public <T> Optional<ResponseEntity<T>> getEntity(String url, Class<T> type, Object... urlVariables) {
+        HttpHeaders headers = createHeaders();
+        HttpEntity<?> entity = new HttpEntity<>(headers);
+        try {
+            ResponseEntity<T> response = this.restTemplate.exchange(
+                    url,
+                    HttpMethod.GET,
+                    entity,
+                    type,
+                    urlVariables
+            );
+            return Optional.of(response);
+        } catch (HttpClientErrorException ex) {
+            if (ex.getStatusCode() != HttpStatus.NOT_FOUND) {
+                throw ex;
+            }
+        }
+        return Optional.empty();
+    }
+
+    private HttpHeaders createHeaders(){
+        Account account = accountService.get();
+        return new HttpHeaders() {{
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the contents of this initializer to a standard constructor or to field initializers. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ecosystem.cloudogu.com/sonar/coding_rules#rule_key=squid%3AS1171)